x-pack/legacy/plugins/siem/public/pages/detection_engine/components/user_info/index.tsx

```diff
@@ -0,0 +1,185 @@
+      };
+    }
+    case 'updateSignalIndexName': {
+      return {
+        ...state,
+        signalIndexName: action.signalIndexName,
+      };
+    }
+    default:
+      return state;
+  }
+};
+
+const StateUserInfoContext = createContext<[State, Dispatch<Action>]>([initialState, () => noop]);
+
+const useUserData = () => useContext(StateUserInfoContext);
+
+interface ManageUserInfoProps {
+  children: React.ReactNode;
+}
+
+export const ManageUserInfo = ({ children }: ManageUserInfoProps) => (
+  <StateUserInfoContext.Provider value={useReducer(userInfoReducer, initialState)}>
+    {children}
+  </StateUserInfoContext.Provider>
+);
+
+export const useUserInfo = (): Return => {
+  const [
+    { canUserCRUD, hasWrite, isSignalIndexExists, isAuthenticated, loading, signalIndexName },
+    dispatch,
+  ] = useUserData();
+  const [privilegeLoading, isApiAuthenticated, hasApiWrite] = usePrivilegeUser();
+  const [
+    indexNameLoading,
+    isApiSignalIndexExists,
+    apiSignalIndexName,
+    createSignalIndex,
+  ] = useSignalIndex();
+
+  const uiCapabilities = useKibana().services.application?.capabilities;
+  const capabilitiesCanUserCRUD = (uiCapabilities?.siem?.crud as boolean) ?? false;
+
+  useEffect(() => {
+    if (loading !== privilegeLoading || indexNameLoading) {
+      dispatch({ type: 'updateLoading', loading: privilegeLoading || indexNameLoading });
+    }
+  }, [loading, privilegeLoading, indexNameLoading]);
+
+  useEffect(() => {
+    if (hasWrite !== hasApiWrite && hasApiWrite != null) {
+      dispatch({ type: 'updateHasWrite', hasWrite: hasApiWrite });
+    }
+  }, [hasWrite, hasApiWrite]);
+
+  useEffect(() => {
+    if (isSignalIndexExists !== isApiSignalIndexExists && isApiSignalIndexExists != null) {
+      dispatch({ type: 'updateIsSignalIndexExists', isSignalIndexExists: isApiSignalIndexExists });
+    }
+  }, [isSignalIndexExists, isApiSignalIndexExists]);
+
+  useEffect(() => {
+    if (isAuthenticated !== isApiAuthenticated && isApiAuthenticated != null) {
+      dispatch({ type: 'updateIsAuthenticated', isAuthenticated: isApiAuthenticated });
+    }
+  }, [isAuthenticated, isApiAuthenticated]);
+
+  useEffect(() => {
+    if (canUserCRUD !== capabilitiesCanUserCRUD && capabilitiesCanUserCRUD != null) {
+      dispatch({ type: 'updateCanUserCRUD', canUserCRUD: capabilitiesCanUserCRUD });
+    }
+  }, [canUserCRUD, capabilitiesCanUserCRUD]);
+
+  useEffect(() => {
+    if (signalIndexName !== apiSignalIndexName && apiSignalIndexName != null) {
+      dispatch({ type: 'updateSignalIndexName', signalIndexName: apiSignalIndexName });
+    }
+  }, [signalIndexName, apiSignalIndexName]);
+
+  useEffect(() => {
+    if (
+      isAuthenticated &&
+      hasApiWrite &&
+      isSignalIndexExists != null &&
+      !isSignalIndexExists &&
+      createSignalIndex != null
+    ) {
+      createSignalIndex();
+    }
+  }, [createSignalIndex, isAuthenticated, isSignalIndexExists, hasWrite]);
```

Comment:
You have `hasWrite` as a dep but in the `if` check you are checking for `hasApiWrite` did you want to use that as a dep down here for your useEffect?